nikola/plugins/template/jinja.py

```diff
@@ -112,7 +112,7 @@ def get_string_deps(self, text):
             filename = self.lookup.loader.get_source(self.lookup, dep_name)[1]
             deps.add(filename)
             sub_deps = self.get_deps(filename)
-            self.dependency_cache[dep_name] = sub_deps
+            self.dependency_cache[dep_name] = [filename] + sub_deps
```

Comment:
Are you sure this is the place? We return `deps` from this function, and it’s not taken from the cache. Perhaps add `[filename]` a line earlier?